scripts/bdc/get_bdc_studies_md_from_gen3.py

```diff
@@ -0,0 +1,269 @@
+import csv
+import json
+import logging
+import os
+import re
+import sys
+import urllib.parse
+import xml.dom.minidom as minidom
+import xml.etree.ElementTree as ETree
+from collections import defaultdict, Counter
+from datetime import datetime
+import requests
+import argparse
+
+# Configuration
+GEN3_DOWNLOAD_LIMIT = 50
+
+class Gen3DataProcessor:
+    def __init__(self, output_dir=None):
+        # Use user-specified output directory
+        self.output_dir = output_dir if output_dir else "gen3_output"
+        self.setup_directory()
+        self.setup_logging()
+        
+    def setup_directory(self):
+        os.makedirs(self.output_dir, exist_ok=True)
+        
+    def setup_logging(self):
+        self.log_file = os.path.join(self.output_dir, "processing.log")
+
+        for handler in logging.root.handlers[:]:
+            logging.root.removeHandler(handler)
+            
+        # Configure logging format
+        logging_format = '%(asctime)s - %(levelname)s - %(message)s'
+        logging.basicConfig(
+            level=logging.INFO,
+            format=logging_format,
+            handlers=[
+                logging.FileHandler(self.log_file),
+                logging.StreamHandler(sys.stdout)
+            ]
+        )
+        self.logger = logging.getLogger(__name__)
+        
+    def download_gen3_list(self, input_url, download_limit=GEN3_DOWNLOAD_LIMIT):
+        complete_list = []
+        offset = 0
+        while True:
+            url = input_url + f"&limit={download_limit}&offset={offset}"
+            self.logger.debug(f"Requesting GET {url} from Gen3")
+            
+            try:
+                partial_list_response = requests.get(url)
+                partial_list_response.raise_for_status()
+                
+                partial_list = partial_list_response.json()
+                complete_list.extend(partial_list)
+                
+                if len(partial_list) < download_limit:
+                    break
+                    
+                offset += download_limit
+                
+            except requests.exceptions.RequestException as e:
+                self.logger.error(f"Error downloading from Gen3: {e}")
+                raise
+        
+        if len(complete_list) != len(set(complete_list)):
+            duplicate_ids = sorted([ident for ident, count in Counter(complete_list).items() if count > 1])
+            self.logger.warning(f"Found duplicate discovery_metadata: {duplicate_ids}")
```

Comment:
```suggestion
            raise ValueError(f"Found duplicate discovery_metadata: {duplicate_ids}")
```

I don't think you're supposed to have duplicate discovery_metadata entries, so I would suggest raising an exception here so you don't miss this.